src/containers/SarcophagusResurrection/index.tsx

```diff
@@ -0,0 +1,138 @@
+    />
+  );
+};
+
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
+
+const hexString = (value: string) => {
+  let hexKey;
+  if (value?.substr(0, 2) !== "0x") hexKey = "0x" + value;
+  return hexKey || value;
+};
+
+interface PrivateKeyFieldProps {
+  handleQuery: (values: any) => void;
+  initialValues: any;
+}
+
+const validationSchema = (hasKeys: boolean) => {
+  if (hasKeys) return;
+  return Yup.object().shape({
+    recipientPrivateKey: Yup.string()
+      .test("required", "Recipient Private Key is required", (value) => !!value)
+      .test("validDataHextString", "Please enter a valid private key", (value) => {
+        let testValue;
+        const str = value?.substr?.(0, 2);
+        if (str !== "0x") testValue = "0x" + value;
+        return utils.isHexString(testValue || value, 32);
+      }),
+  });
+};
+
+const PrivateKeyField = ({ handleQuery, initialValues }: PrivateKeyFieldProps) => {
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={(values) => handleQuery(values)}
+    >
+      {({ values, errors, handleChange, handleSubmit }) => (
+        <form onSubmit={handleSubmit} className="px-2 w-128">
+          <RecipientPrivateKeyField
+            values={values}
+            errors={errors}
+            handleChange={handleChange}
+            isVisible={true}
+          />
+          <Button addClasses="mb-4" width="full" type="submit" label="Submit" />
+        </form>
+      )}
+    </Formik>
+  );
+};
+
+const SarcophagusResurrection = () => {
+  // retrieve keys from query
+  let query = useQuery();
+  const [recipientPrivateKey, setRecipientPrivateKey] = useState("");
+
+  // takes private key param and converts to address
+  const address = recipientPrivateKey ? utils.computeAddress(hexString(recipientPrivateKey)) : "";
+  const { allRecipientSarcophagi, isRecipientSarcophagiLoaded } = useRecipient(address, true);
+
+  const initialValues = {
+    recipientPrivateKey: query.get("recipientPrivateKey") || "",
+  };
+
+  const handleQuery = (values: any) => {
+    setRecipientPrivateKey(values.recipientPrivateKey);
+  };
+
+  return (
+    <div className="pt-8 px-8 flex justify-center md:justify-between flex-wrap md:flex-nowrap gap-4 md:gap-8">
+      <div style={{ width: "34rem" }}>
+        <div className={Heading.PageHeading}>
+          <img src={icon} alt="" className="mr-4" />
+          <span>Resurrection</span>
+        </div>
+        <div className="text-md font-normal pl-2 pt-8 pb-4 leading-6">
+          If the embalmer does not attest to their Sarcophagus, this is where the receiver will type in the
+          private key to decrypt the inner layer and gain access to the file.
+        </div>
+        <PrivateKeyField handleQuery={handleQuery} initialValues={initialValues} />
+      </div>
+      <div className="" style={{ width: "34rem" }}>
+        {!isRecipientSarcophagiLoaded && recipientPrivateKey ? (
```

Comment:
What happens if `isRecipientSarcophagiLoaded` is `true`, but `recipientPrivateKey` is null? Should this be:
`!isRecipientSarcophagiLoaded || !recipientPrivateKey`?

Reply:
Or do we want to display an error message if the `recipientPrivateKey` is null, b/c that means the query value in URL wasn't set?